framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -205,6 +225,33 @@ public void updateForMethodCall(
             MethodInvocationNode n, AnnotatedTypeFactory atypeFactory, V val) {
         ExecutableElement method = n.getTarget().getMethod();
 
+        // List of expressions that this method side effects (specified as annotation values of
+        // @SideEffectsOnly). If the value is null, then there is no @SideEffectsOnly annotation.
```

Comment:
How can the value be null?  Here it is set to a non-null value and I don't see it set to null anywhere.

If the code has a bug in that it does not distinguish the cases, please add a test that exposes the bug before fixing it.

Also be sure to have tests that include code annotated with `@SideEffectsOnly({})`.